### PR TITLE
[ONL-6716] - Migrate modal component

### PR DESCRIPTION
**Pull request by DisegnAle**
migrated component

### CODE REVIEW

**Review by wolandec — APPROVED** [review]

File: src/components/ec-modal/ec-modal.vue [hunk 1]
```diff
@@ -99,132 +99,123 @@
...
 </template>
 
-<script>
-import EcFocusTrap from '../../directives/ec-focus-trap';
+<script setup>
+import {
+  computed, onBeforeUnmount, ref, useSlots, watchEffect,
+} from 'vue';
+
+import vEcFocusTrap from '../../directives/ec-focus-trap';
```
Comment (mcibique) [inline src/components/ec-modal/ec-modal.vue:107]:
```suggestion
import VEcFocusTrap from '../../directives/ec-focus-trap';
```

File: src/components/ec-modal/ec-modal.vue [hunk 2]
```diff
@@ -99,132 +99,123 @@
...
+}
+
+watchEffect(() => {
+  const value = props.show;
+  if (value) {
+    document.addEventListener('keyup', escapeIsPressed);
+  } else {
+    document.removeEventListener('keyup', escapeIsPressed);
+  }
+});
```
Comment (mcibique) [inline src/components/ec-modal/ec-modal.vue:218]: Similar thing as in phone-number-input, we should group logic together, now when we don't have to follow a structure of options API.

```
// keyboard nav
watchEffect(() => {
onBeforeUnmount(() => {
  document.removeEventListener('keyup', escapeIsPressed);
});
function escapeIsPressed(e) {
function closeModal() {

// buttons
const positiveButton
const negativeButton
const isLoadingPositiveButton
const isLoadingNegativeButton
const positiveButtonCategory
const negativeButtonCategory
function hasPositiveButton() {
function hasNegativeButton() {

// focus trap
getFocusTrapOptions

// footer
function hasFooterLeftContent() {
function hasFooter() {

// z-index
const zIndexStyle 
```

File: src/components/ec-modal/ec-modal.vue [hunk 3]
```diff
@@ -99,132 +99,123 @@
...
+  document.removeEventListener('keyup', escapeIsPressed);
+});
+
+function negativeAction() {
+  emit('negative');
+}
+
+function positiveAction() {
+  emit('positive');
+}
```
Comment (mcibique) [inline src/components/ec-modal/ec-modal.vue:165]: You could inline negativeAction and positiveAction into the template.

File: src/components/ec-modal/ec-modal.vue [hunk 4]
```diff
@@ -99,132 +99,123 @@
...
+const negativeButton = ref(null);
+
+const isLoadingPositiveButton = computed(() => !!props.isLoading.positive);
+const isLoadingNegativeButton = computed(() => !!props.isLoading.negative);
+const positiveButtonCategory = computed(() => props.category.positive || 'primary');
+const negativeButtonCategory = computed(() => props.category.negative || 'secondary');
+const zIndexStyle = computed(() => (props.zIndex ? { zIndex: props.zIndex } : null));
+
+onBeforeUnmount(() => {
+  document.removeEventListener('keyup', escapeIsPressed);
```
Comment (mcibique) [inline src/components/ec-modal/ec-modal.vue:156]: I'd rename the function `escapeIsPressed` to `onKeyUp`. The function is not executed when the escape is pressed but when the key gets released and it's checking whether the ESC got pressed inside.

**Review by mcibique — APPROVED** [review]

**Pull request merged**